string goal_name
---
bool success